src/cobra/core/reaction.py

```diff
@@ -999,13 +996,11 @@ def check_mass_balance(self):
     @property
     def compartments(self):
         """lists compartments the metabolites are in"""
-        if self._compartments is None:
-            self._compartments = {
-                met.compartment
-                for met in self._metabolites
-                if met.compartment is not None
-            }
-        return self._compartments
+        return {
+            met.compartment
+            for met in self._metabolites
+            if met.compartment is not None
+        }
```

Comment:
Sorry, my bad. This is what the formatter wants now.

```suggestion
        return {
            met.compartment for met in self._metabolites if met.compartment is not None
        }
```